### PR TITLE
chore: Add "no-nested-ternary" ESLint rule to projects

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What?
- Adds "no-nested-ternary" rules to ESLint configs ([docs](https://eslint.org/docs/latest/rules/no-nested-ternary))
- Refactors nested ternaries to satisfy rule

## Why?
- I saw [this comment](https://github.com/theopensystemslab/planx-new/pull/3483/files#r1715103291) on #3483 and agreed - as we grow as a team we likely want to move more towards codifying these conventions we aim to follow in the codebase
- Hopefully some of the "before" snippets refactored in this PR demonstrate that nested ternaries are actually very hard to follow - understanding some of these took a surprising amount of mental capacity for this time of day 😅

### CODE REVIEW

File: editor.planx.uk/src/@planx/components/Checklist/Editor.tsx
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/Checklist/Editor.tsx:1]: There a few changes to the `Checklist` component due to the fact that it's props accept `options` and `groupedOptions` (both optional). In reality only one can ever be present so a wider refactor to make the props reflect this reality might simplify a few things here.

**Review by jessicamcinchak — COMMENTED** [review]
:heavy_plus_sign:

File: editor.planx.uk/src/@planx/components/PlanningConstraints/Public.tsx [hunk 1]
```diff
@@ -146,98 +146,100 @@ function Component(props: Props) {
...
+        props.handleSubmit?.({
+          data: passportData,
+        });
+      }}
+      refreshConstraints={() => mutate()}
+      inaccurateConstraints={inaccurateConstraints}
+      setInaccurateConstraints={setInaccurateConstraints}
+    />
+  )
+
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/@planx/components/PlanningConstraints/Public.tsx:242]: Thanks for `PlanningConstraints` refactors here, I was very much the culprit of this gnarly one that scaled very poorly as we added complexity to this component :sweat_smile:

**Review by jessicamcinchak — COMMENTED** [review]
:raised_hands:

**Review by jessicamcinchak — APPROVED** [review]
